akka-stream/src/main/scala/akka/stream/javadsl/Tcp.scala

```diff
@@ -84,41 +84,42 @@ object StreamTcp extends ExtensionId[StreamTcp] with ExtensionIdProvider {
     def localAddress: InetSocketAddress = delegate.localAddress
   }
 
-  override def get(system: ActorSystem): StreamTcp = super.get(system)
+  override def get(system: ActorSystem): Tcp = super.get(system)
 
-  def lookup() = StreamTcp
+  def lookup() = Tcp
 
-  def createExtension(system: ExtendedActorSystem): StreamTcp = new StreamTcp(system)
+  def createExtension(system: ExtendedActorSystem): Tcp = new Tcp(system)
 }
 
-class StreamTcp(system: ExtendedActorSystem) extends akka.actor.Extension {
-  import StreamTcp._
+class Tcp(system: ExtendedActorSystem) extends akka.actor.Extension {
+  import Tcp._
   import akka.dispatch.ExecutionContexts.{ sameThreadExecutionContext ⇒ ec }
 
-  private lazy val delegate: scaladsl.StreamTcp = scaladsl.StreamTcp(system)
+  private lazy val delegate: scaladsl.Tcp = scaladsl.Tcp(system)
 
   /**
-   * Creates a [[StreamTcp.ServerBinding]] instance which represents a prospective TCP server binding on the given `endpoint`.
+   * Creates a [[Tcp.ServerBinding]] instance which represents a prospective TCP server binding on the given `endpoint`.
    */
-  def bind(endpoint: InetSocketAddress,
+  def bind(interface: String,
+           port: Int,
            backlog: Int,
            options: JIterable[SocketOption],
            idleTimeout: Duration): Source[IncomingConnection, Future[ServerBinding]] =
-    Source.adapt(delegate.bind(endpoint, backlog, immutableSeq(options), idleTimeout)
+    Source.adapt(delegate.bind(interface, port, backlog, immutableSeq(options), idleTimeout)
       .map(new IncomingConnection(_))
       .mapMaterialized(_.map(new ServerBinding(_))(ec)))
 
   /**
-   * Creates a [[StreamTcp.ServerBinding]] without specifying options.
+   * Creates a [[Tcp.ServerBinding]] without specifying options.
    * It represents a prospective TCP server binding on the given `endpoint`.
    */
-  def bind(endpoint: InetSocketAddress): Source[IncomingConnection, Future[ServerBinding]] =
-    Source.adapt(delegate.bind(endpoint)
+  def bind(interface: String, port: Int): Source[IncomingConnection, Future[ServerBinding]] =
```

Comment:
Why not keep both? `bind(addr)` and `bind("", 23)`? In case I happen to have an InetSocketAddress from somewhere